src/test/java/org/apache/sysds/test/functions/federated/codegen/FederatedRowwiseTmplTest.java

```diff
@@ -0,0 +1,181 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sysds.test.functions.federated.codegen;
+
+import java.io.File;
+import org.apache.sysds.common.Types.ExecMode;
+import org.apache.sysds.runtime.matrix.data.MatrixValue.CellIndex;
+import org.apache.sysds.runtime.meta.MatrixCharacteristics;
+import org.apache.sysds.runtime.util.HDFSTool;
+import org.apache.sysds.test.AutomatedTestBase;
+import org.apache.sysds.test.TestConfiguration;
+import org.apache.sysds.test.TestUtils;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+
+@RunWith(value = Parameterized.class)
+@net.jcip.annotations.NotThreadSafe
+public class FederatedRowwiseTmplTest extends AutomatedTestBase
+{
+	private final static String TEST_NAME = "FederatedRowwiseTmplTest";
+
+	private final static String TEST_DIR = "functions/federated/codegen/";
+	private final static String TEST_CLASS_DIR = TEST_DIR + FederatedRowwiseTmplTest.class.getSimpleName() + "/";
+
+	private final static String TEST_CONF = "SystemDS-config-codegen.xml";
+
+	private final static String OUTPUT_NAME = "Z";
+	private final static double TOLERANCE = 1e-13;
+	private final static int BLOCKSIZE = 1024;
+
+	@Parameterized.Parameter()
+	public int test_num;
+	@Parameterized.Parameter(1)
+	public int rows;
+	@Parameterized.Parameter(2)
+	public int cols;
+
+	@Override
+	public void setUp() {
+		addTestConfiguration(TEST_NAME, new TestConfiguration(TEST_CLASS_DIR, TEST_NAME, new String[]{OUTPUT_NAME}));
+	}
+
+	@Parameterized.Parameters
+	public static Collection<Object[]> data() {
+		// rows must be even
+		return Arrays.asList(new Object[][] {
+			// {test_num, rows, cols}
+
+			{1, 6, 4},
+			{2, 6, 2},
+			{3, 6, 4},
+			{4, 6, 4},
+			{10, 150, 10},
+			{15, 150, 10},
+			{20, 1500, 8},
+			{21, 1500, 8},
+			{25, 600, 10},
+			{31, 150, 10},
+			{40, 300, 20},
+			{45, 1500, 100},
+			{50, 376, 4},
+		});
+	}
+
+	@BeforeClass
+	public static void init() {
+		TestUtils.clearDirectory(TEST_DATA_DIR + TEST_CLASS_DIR);
+	}
+
+	// @Test
+	// public void federatedCodegenRowwiseSingleNode() {
+	// 	testFederatedCodegenRowwise(ExecMode.SINGLE_NODE);
+	// }
+	// 
+	// @Test
+	// public void federatedCodegenRowwiseSpark() {
+	// 	testFederatedCodegenRowwise(ExecMode.SPARK);
+	// }
```

Comment:
Have you verified if these work? or just copied them?
if the latter, please try to execute them, and then enable them if it works, if they don't work add an @Ignore clause to them such that they show up as tests that currently are skipped. Or try to fix it so that they work.

Reply:
Both outcommented tests (SingleNode and Spark) are working, I just disabled them to preserve the test performance. We did the same for the cellwise tests, but I will enable them again.